recipe/meta.yaml

```diff
@@ -21,10 +21,11 @@ source:
     folder: testing
 
 build:
-  number: 6
+  number: 7
   # for cuda support, building with one version is enough to be compatible with
   # all later versions, since arrow is only using libcuda, and not libcudart.
-  skip: true  # [cuda_compiler_version not in ("None", cuda_compiler_version_min)]
+  # this is true despite building with the latest CUDA version.
+  skip: true  # [cuda_compiler_version != "None" and cuda_compiler_version != cuda_compiler_version_min]
```

Comment:
We're actually still building with 10.2 / 11.2 here - is that intentional? Otherwise, I don't see what this PR achieves (aside from resolving how we depend on the cuda drivers). This feedstock is special compared to most others, in that it doesn't _need_ to be rebuilt with CUDA 12 to be usable with that. 

I continue to be amazed by the apparent fact that - contrary to pretty much everything else - we're apparently able to build this feedstock on a newer CUDA version than we need at runtime. (You confirmed my suggestion "this is true despite building with the latest CUDA version", and this also matches Keith's statements on this topic w.r.t. depending only on `__cuda >=9` despite building on 10.2/11.2)

So if that's the case, how about we explicitly move the build to CUDA 12, but have a run-dependence on `__cuda >9`?

Reply:
That's my understanding of what was being done here before (and what is expected by the surrounding discussion in this PR and related issues). This is merely consistent with that

Perhaps I'm misunderstanding what is desired here?

Would be good if @kkraus14 also weighed in

Reply:
The `>=9` constraint may actually be incorrect here. Arrow will happily build and run with the CUDA 9.0 driver, but I'm not sure if it's valid to build it with the CUDA 10.2 driver and run it on the CUDA 9.0 driver.

The `libcuda.so` shared object doesn't version its `SONAME` so there wouldn't be a failure to find the shared library, but I don't believe there's any guarantees that things won't change under the covers in a way that makes it incompatible.

To be safe we should probably match the `__cuda` constraint to `>= cuda_compiler_version`.

Reply:
> but I'm not sure if it's valid to build it with the CUDA 10.2 driver and run it on the CUDA 9.0 driver.

Ah, finally I feel like I understand the world again! 😅 

Well, in that case, we should definitely stay on `cuda_compiler_version_min`, and use that as a lower bound.

So can we agree on the following?
```
      run: 
        - __cuda >= {{ cuda_compiler_version_min }}   # [cuda_compiler_version != "None"]
      run_constrained:
        - apache-arrow-proc =*={{ build_ext }}
        # IIUC we should be able to delete the following, as they
        # should be implied by the run-constraint on __cuda
        # - cuda-version >={{ cuda_compiler_version_min }}  # [cuda_compiler_version != "None"]
        # - cudatoolkit >={{ cuda_compiler_version_min }}   # [cuda_compiler_version != "None"]
```

Reply:
Rather than changing the skip condition here, I'd rather do the following: https://github.com/conda-forge/arrow-cpp-feedstock/commit/1f774a98d8a6fd1dc4d660aaf5ce0d8c7ae95ff9

Reply:
That looks good to me!

Reply:
Could we please move the `run` discussion into the above thread on that topic ( https://github.com/conda-forge/arrow-cpp-feedstock/pull/1079#discussion_r1222345851 )?

Would also like to confirm that the caveats mentioned in this comment ( https://github.com/conda-forge/arrow-cpp-feedstock/pull/1079#discussion_r1222353879 ) (and reference) were understood before making changes

Sorry to be pedantic, but there's already been some confusion on the use of `run`  with virtual packages and would like to avoid further confusion

Thanks for understanding and helping finalize this work! 🙏